### DISCUSSION TITLE
<title> I want to know how to use Excel macro in C#

### DISCUSSION THREAD

**Discussion by whereismylife** [community/community#44471] [Discussions] [answered]
### Select Topic Area

Product Feedback, C#

### Body

Hi. 

recently,  I started making a program a week ago. I was interested in making a program, but recently I started to get into trouble. 
I made a button to open the Excel file, then filled in the value using the text box, and then pressed the save button to make a single csv file.But what I want more is to press the button to open the file and then automatically put it in the text box to create one Excel file.
What are the functions to solve this? Or it would be good to tell them the code or method! for the record, I am making a program in the c# environment.

**Comment by jge162** [comment, marked as answer]
In order to automatically open an Excel file and populate a text box with its contents, you can use the Microsoft.Office.Interop.Excel library to interact with Excel files in C#.

**Reply by whereismylife** [reply]
> Excel 파일을 자동으로 열고 텍스트 상자를 해당 내용으로 채우려면 Microsoft.Office.Interop.Excel 라이브러리를 사용하여 C#의 Excel 파일과 상호 작용할 수 있습니다.

Then, is it possible to automatically extract it as an Excel file by pressing the button?

**Reply by jge162** [reply]
Here is code example 

   using Microsoft.Office.Interop.Excel;
using System.Data //…

private void openFileButton_Click(object sender, EventArgs e) {
    // Open a file dialog to select the Excel file
    OpenFileDialog openFileDialog = new OpenFileDialog();
    openFileDialog.Filter = "Excel files (*.xlsx)|*.xlsx|All files (*.*)|*.*";
    openFileDialog.FilterIndex = 1;
    openFileDialog.RestoreDirectory = true;

    if (openFileDialog.ShowDialog() == DialogResult.OK) {
        // Open the selected Excel file
        Application excelApp = new Application();
        Workbook excelWorkbook = excelApp.Workbooks.Open(openFileDialog.FileName);
        Worksheet excelWorksheet = excelWorkbook.Sheets[1];

        // Read the contents of the Excel file into a DataTable
        Range excelRange = excelWorksheet.UsedRange;
        object[,] excelData = excelRange.Value2;
        DataTable dataTable = new DataTable();
        for (int i = 1; i <= excelRange.Columns.Count; i++) {
            dataTable.Columns.Add();
        }
        for (int i = 1; i <= excelRange.Rows.Count; i++) {
            DataRow dataRow = dataTable.NewRow();
            for (int j = 1; j <= excelRange.Columns.Count; j++) {
                dataRow[j - 1] = excelData[i, j];
            }
            dataTable.Rows.Add(dataRow);
        }

        // Display the contents of the DataTable in the text box
        textBox.Text = dataTable.ToString();

        // Close the Excel file
        excelWorkbook.Close();
        excelApp.Quit();
    }
}

**Reply by whereismylife** [reply]
thank you so much! I will study by opening each code. Thank you for letting me know.

**Comment by jge162** [comment]
Yes
